core/services/assets/Registry.php

```diff
@@ -382,16 +383,36 @@ public function registerManifestFile($namespace, $url_base, $manifest_file)
             );
         }
         if (filter_var($url_base, FILTER_VALIDATE_URL) === false) {
-            throw new InvalidArgumentException(
-                sprintf(
-                    esc_html__(
-                        'The provided value for %1$s is not a valid url.  The url provided was: %2$s',
-                        'event_espresso'
+            if ( WP_DEBUG ) {
+                throw new InvalidArgumentException(
+                    sprintf(
+                        esc_html__(
+                            'The provided value for %1$s is not a valid url.  The url provided was: %2$s',
+                            'event_espresso'
+                        ),
+                        '$url_base',
+                        $url_base
+                    )
+                );
+            }
+            if (is_admin()) {
+                EE_Error::add_error(
+                    sprintf(
+                        esc_html__(
+                            'The url given for %1$s assets is invalid.  The url provided was: "%2$s". This usually happens when another plugin or theme on a site is using the "%3$s" filter or has an invalid url set for the "%4$s" constant',
```

Comment:
why not use this same message in the exception? I think it would be helpful for developers to know the possible cause.

Reply:
Ya I could I suppose.  The original message was based on the (incorrect) assumption that `plugins_dir_url` (via `EventEspresso\core\domain\Domain::distributionAssetsUrl`) would always return a valid url.  In the case where `registerManifestUrl` is being called by third party (or add-on) code to register a manifest, it'd be pretty clear that the provided url is incorrect.

So... the extra added here is mostly for non developers encountering this error in the wild in which case it's more _likely_ the cause is as described.